src/network/modules/ledger/ledger.ts

```diff
@@ -7,15 +7,82 @@ export interface LedgerInfo {
   symbols: Map<ReturnType<Identity["toString"]>, string>
 }
 
+export enum OrderType {
+  indeterminate = 0,
+  ascending = 1,
+  descending = 2,
+}
+
+interface RangeBound {
+  boundType: BoundType
+  value: unknown
+}
+
+interface TxnRangeBound extends Omit<RangeBound, "value"> {
+  value: Uint8Array
+}
+
+interface ListArgs {
+  count?: number
+  filters?: ListFilterArgs
+  order?: OrderType
+}
+
+export enum TransactionType {
+  send = "send",
+  burn = "burn",
+  mint = "mint",
+}
+export interface Transaction {
+  id: string
+  time: Date
+  type: TransactionType
+  amount: bigint
+  symbolIdentity: string
+  symbol?: string
+  from?: string
+  to?: string
+  account?: string
+}
+
+interface TransactionsData {
+  count: number
+  transactions: Transaction[]
+}
+
+export enum BoundType {
+  "unbounded" = "unbounded",
+  "inclusive" = "inclusive",
+  "exclusive" = "exclusive",
+}
+
+export interface ListFilterArgs {
+  accounts?: string | string[]
+  symbols?: string | string[]
+  txnIdRange?: [TxnRangeBound?, TxnRangeBound?]
+}
+
+interface ListFilters {
+  0: string | string[]
+  1: number | number[]
+  2: string | string[]
+  3: Map<number, [(0 | 1)?, Uint8Array?]>
```

Comment:
`export type Range<T> = [T] | [0, T] | [1, T];`

Reply:
we have removed `ListFilters` and added `Bound` and `Range`.  `Range` is `type Range<T> = Map<number, Bound<T>>`